plugins/field-colour-hsv-sliders/src/field_colour_hsv_sliders.ts

```diff
@@ -0,0 +1,639 @@
+   * saturation, and brightness parameters, into a hexadecimal string in the
+   * format "#rrggbb".
+   * @param hue The hue of the colour.
+   * @param saturation The saturation of the colour.
+   * @param brightness The brightness of the colour.
+   * @return A hexadecimal representation of the colour in the format "#rrggbb"
+   */
+  private static hsvToHex(
+      hue: number, saturation: number, brightness: number): string {
+    FieldColourHsvSliders.helperHsv.h = hue;
+    FieldColourHsvSliders.helperHsv.s = saturation;
+    FieldColourHsvSliders.helperHsv.v = brightness;
+    return FieldColourHsvSliders.helperRgb.loadFromHsv(
+        FieldColourHsvSliders.helperHsv).toHex();
+  }
+
+  /**
+   * Updates the value of this field based on the editor sliders.
+   * @param event Unused.
+   */
+  private onSliderChange(event?: Event): void {
+    if (!this.hueSlider || !this.saturationSlider || !this.brightnessSlider) {
+      throw new Error('The HSV sliders are missing.');
+    }
+    const hue: number = parseFloat(this.hueSlider.value) /
+        FieldColourHsvSliders.HUE_SLIDER_MAX;
+    const saturation: number = parseFloat(this.saturationSlider.value) /
+        FieldColourHsvSliders.SATURATION_SLIDER_MAX;
+    const brightness: number = parseFloat(this.brightnessSlider.value) /
+        FieldColourHsvSliders.BRIGHTNESS_SLIDER_MAX;
+    this.setValue(FieldColourHsvSliders.hsvToHex(
+        hue, saturation, brightness));
+    this.renderSliders();
+  }
+
+  /**
+   * Updates the value of this field and editor sliders using an eyedropper.
+   * @param event Unused.
+   */
+  private onEyedropperEvent(event?: Event): void {
+    if (window.EyeDropper) {
+      const eyeDropper: EyeDropper = new window.EyeDropper();
+      eyeDropper.open().then((result) => {
+        this.setValue(result.sRGBHex);
+        this.updateSliderValues();
+      });
+    }
+  }
+
+  /**
+   * Updates the gradient backgrounds of the slider tracks and readouts based
+   * on the slider values.
+   */
+  private renderSliders(): void {
+    if (!this.hueSlider || !this.hueReadout ||
+        !this.saturationSlider || !this.saturationReadout ||
+        !this.brightnessSlider || !this.brightnessReadout) {
+      throw new Error('The HSV sliders are missing.');
+    }
+    this.hueReadout.textContent = this.hueSlider.value;
+    this.saturationReadout.textContent = this.saturationSlider.value;
+    this.brightnessReadout.textContent = this.brightnessSlider.value;
+
+    const h: number = parseFloat(this.hueSlider.value) /
+        FieldColourHsvSliders.HUE_SLIDER_MAX;
+    const s: number = parseFloat(this.saturationSlider.value) /
+        FieldColourHsvSliders.SATURATION_SLIDER_MAX;
+    const v: number = parseFloat(this.brightnessSlider.value) /
+        FieldColourHsvSliders.BRIGHTNESS_SLIDER_MAX;
+
+    // The hue slider needs intermediate gradient control points to include all
+    // colours of the rainbow.
```

Comment:
Up to you, but this might be a bit cleaner if the whole thing was made a template literal? I could see that potentially being worse as well though...

Reply:
There's a bunch of expression parsing in here, and this method isn't creating any new elements anyway, so I'll leave it the way it is.